alembic/versions/9359297269a8_convert_ribbon_to_a_multichoice_column.py

```diff
@@ -0,0 +1,64 @@
+"""Convert ribbon to a multichoice column.
+
+Revision ID: 9359297269a8
+Revises: 167243c0e86c
+Create Date: 2017-07-21 09:18:38.444238
+
+"""
+
+
+# revision identifiers, used by Alembic.
+revision = '9359297269a8'
+down_revision = '167243c0e86c'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import table
+
+
+try:
+    is_sqlite = op.get_context().dialect.name == 'sqlite'
+except:
+    is_sqlite = False
+
+if is_sqlite:
+    op.get_context().connection.execute('PRAGMA foreign_keys=ON;')
+    utcnow_server_default = "(datetime('now', 'utc'))"
+else:
+    utcnow_server_default = "timezone('utc', current_timestamp)"
+
+# We need this table in order to upgrade/downgrade the ribbon column
+attendee = table(
+    'attendee',
+    sa.Column('ribbon', sa.Unicode())
+)
+
+def upgrade():
+    op.alter_column('attendee', 'ribbon', type_=sa.Unicode(), server_default = '', nullable = False)
+
+    op.execute(
+        attendee
+            .update()
+            .where(attendee.c.ribbon == '154973361')
+            .values({'ribbon': ''})
+    )
+
+
+def downgrade():
+    op.execute(
+        attendee
+            .update()
+            .where(attendee.c.ribbon == '' or attendee.c.ribbon.contains(','))
+            .values({'ribbon': 154973361})
```

Comment:
This handles both:
1. Downgrading when people have multiple ribbons
2. Downgrading a blank ribbon to c.NO_RIBBON

We convert the attendee's ribbon to NO_RIBBON in both cases, which wouldn't be good for a production system, but as you noted earlier we don't want to downgrade on a production system. I did, however, need to downgrade on my dev box in order to test the new upgrade.